### ISSUE TITLE
[BUG] Quest description removes secret block for GM

### ISSUE THREAD

**Issue by atnoslen** [League-of-Foundry-Developers/foundryvtt-forien-quest-log#107]
**Module Version:** v0.7.5

**Describe the bug**
When using the editor in the description, if you add a secret block, it is removed during display for the GM.  The relevant sections are back upon edit, but are missing on display.

Reason: pre-populate description dialog to unlock.  There is GM Notes, but being able to pre-position text to unhide can be handy.

As GM, visible:
![image](https://user-images.githubusercontent.com/3331625/126840964-426c8e14-77fc-457d-88a7-94fccb303415.png)

As GM, edited (pardon PF2E journal editor...):
![image](https://user-images.githubusercontent.com/3331625/126841016-a7d19c18-e07c-436c-aa38-a32878f89d09.png)

To unhide, I wrote a script that removes the `secret` class from the section by an element id (only relevant section shown).

```
const questId = args[0];
const dialogId = args[2];

const QuestAPI = game.modules.get('forien-quest-log').public.QuestAPI;

const quest = QuestAPI.DB.getQuest(questId);

let content = $('<div>' + quest.description + '</div>');
$('#'+dialogId, content).removeClass('secret');

quest.description = content.html();
```

Note, I cannot edit the HTML of the description directly and must set programmatically (next issue to submit).

**To Reproduce**
1. Create quest
2. Edit quest description
3. Type anything
4. Change to `secret`
5. Save changes

**Expected behavior**
Secret text is rendered for GM, highlighted via `.secret` CSS.

**Screenshots**
See above

**Browser:**
Firefox

**Foundry Version:** 
0.8.8

**Game System:** 
PF2E 2.8.1.6593

**Comment by typhonrt** [comment]
Hi @atnoslen. Indeed the display of the secret block is not showing correctly when in non-edit display / mode for those that can edit quests. I'll have a fix in shortly and further comment on this issue soon for an improvement to your macro. It should be noted that the secret block is not intended to work how you are using it, but this is a neat hack nonetheless!

The good thing is that your comment and macro example helped improve the security of FQL by bringing awareness to the implications of having the QuestDB exposed publicly. I'm now using `DOMPurify` to sanitize all display of template data where user input can be involved. I prevented these issues through the GUI, but indeed with public access to the QuestDB direct manipulation certainly is possible. This is handy as it allows the re-introduction of the source code editing plugin to the TinyMCE editor and also entity content linking being enabled in quest objectives and abstract rewards. So all in all thanks for posting this issue!

**Comment by typhonrt** [comment]
Hi @atnoslen. This is fixed and will be released later this week in `0.7.6`. The code editor is enabled again, so you can do the editing to add an ID. You could also try using an [nth-child JQuery selector](https://api.jquery.com/nth-child-selector/) for section, but I suppose you macro could more easily target sections with the approach you have already as unique names may be easier to manage if you have multiple secret sections. 

Anyway thanks for submitting the issue!

**Comment by typhonrt** [comment]
Released with `0.7.6` which is available now.
